atomix/cluster/src/test/java/io/atomix/raft/RaftFailOverIT.java

```diff
@@ -379,34 +379,6 @@ public void shouldTruncateLogOnNewerSnapshot() throws Throwable {
     assertThat(entries.get(0).index()).isEqualTo(201);
   }
 
-  @Test
-  public void shouldTruncateLogOnNewerSnapshotEvenAfterRestart() throws Throwable {
-    // given
-    raftRule.appendEntries(50);
-    final var followerB = raftRule.shutdownFollower();
-    raftRule.appendEntries(250);
-    raftRule.takeCompactingSnapshot(200); // take into account the replication threshold
-    // Leader and Follower A Log [200-300].
-    // Follower B Log [1-51]
-
-    // Follower B has old log AND snapshot
-    final var nodes = raftRule.getNodes();
-    final var followerA = nodes.stream().findFirst().orElseThrow();
-    raftRule.copySnapshotOffline(followerA, followerB);
```

Comment:
🤔 in your PR/commit description you meant it was not working because it was restarting without snapshot, but here we copied one right do mimik the scenario that we were interrupted between snapshot replication/taking and resetting the log correct? This is nothing we need to consider now? As far as I have seen the previous code you changed regarding the listener there was also compaction done after persisting the snapshot ? 🤔

Reply:
It was copying it, and then when it restarted the server was discarding the snapshot store, and thus discarding the snapshot it had just copied :smile: